### ISSUE TITLE
MEDIUM: [$500] Unread indicator is showing in the fevicon even though there is nothing in LHN

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#31276]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:**
**Reproducible in staging?:** needs reproduction
**Reproducible in production?:** needs reproduction
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @puneetlath 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1699899029212629

## Action Performed:
1. Open app and login
2. Make a IOU request 
3. Open the IOU thread to see the details
4. Observe the unread indicator in expensify fevicon

## Expected Result:
If there is an unread indicator, LHN should have one unread message

## Actual Result:
But nothing in LHN with unread message. 

it was caused by this transaction thread, which wasn’t showing up in my LHN, but was considered unread by the logic being used to compute the [unread indicator here](https://github.com/Expensify/App/blob/f23efa1118f52a953dc2b109875c33b3fd206a06/src/libs/UnreadIndicatorUpdater/index.js#L11). We should update the logic of the unread indicator to make sure it matches the logic of the LHN (or even better use shared logic). So that it’s never possible to see the unread indicator when you don’t have anything unread in the LHN.

## Workaround:
unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

![IMG_0222](https://github.com/Expensify/App/assets/38435837/a0169f54-c73c-4fa1-9de4-bb600073f4ed)

![image (31)](https://github.com/Expensify/App/assets/38435837/c9fd048d-25f5-46c1-9dab-af8c3096d299)

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01d98ea368e18737bf</li>
        <li>Upwork Job ID: 1724168903941550080</li>
        <li>Last Price Increase: 2024-01-03</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @muttmuure</details>

**Comment by ericuldall** [comment]
# Proposal
This seems to be an issue with new reports coming in from onyx. There is no check if the change was triggered by the currently logged in user. So subscribed actions from that user will show as unread.

To fix this, we should add an additional check to verify the report was not triggered by the currently logged in user.

**Comment by paultsimura** [comment]
@puneetlath could you please double-check this was not happening simply because of the Concierge chat being unread? This can be easily overseen for the new accounts.

**Comment by AmjedNazzal** [comment]
I think this should be held until [this](https://github.com/Expensify/App/issues/28536#issuecomment-1773926403) is completed because it will apply changes to `UnreadIndicatorUpdater`

**Comment by cubuspl42** [comment]
I think that this issue is a possible duplicate of https://github.com/Expensify/App/issues/28536, or at least has a high chance of getting solved by the associated PR

**Comment by puneetlath** [comment]
@cubuspl42 indeed it does sound very similar to that. @muttmuure perhaps we can put it on hold for #31276 that and then we can have @shawnborton and others who are experiencing this re-test after #31276 is live.

**Comment by puneetlath** [comment]
Lol whops, I meant https://github.com/Expensify/App/issues/28536

**Comment by cubuspl42** [comment]
In case this doesn't move forward, please ping also in PR https://github.com/Expensify/App/pull/30133, as Melvin won't remind me about that issue anymore

**Comment by MonilBhavsar** [comment]
Commented in the PR

**Comment by cubuspl42** [comment]
[Ongoing discussion](https://github.com/Expensify/App/pull/30133#issuecomment-1818960844) on the issue found in the PR

**Comment by MonilBhavsar** [comment]
PR is being reviewed

**Comment by situchan** [comment]
Not overdue

**Comment by MonilBhavsar** [comment]
Discussion is going on PR, looks like it has halted cc @cubuspl42

**Comment by cubuspl42** [comment]
We're waiting for a review by @Gonals 🙂

**Comment by cubuspl42** [comment]
https://github.com/Expensify/App/pull/30133 is merged! You're good to go 🙂 Start with re-testing.

**Comment by MonilBhavsar** [comment]
PR is merged and awaiting deploy

**Comment by cubuspl42** [comment]
Just so we're on the same page, we're talking about the PR associated with the issue we're holding on here 🙂 This one can be put off hold

**Comment by muttmuure** [comment]
I think we can take this off hold now? @situchan what do you think of @dukenv0307's [proposal here](https://github.com/Expensify/App/issues/31276#issuecomment-1809346822)?

**Comment by situchan** [comment]
> We should check for this case the same way we do in `OptionRowLHN`

@dukenv0307's proposal sounds good. But I am still testing with various accounts to validate it.

**Comment by MonilBhavsar** [comment]
Proposal being tested

**Comment by muttmuure** [comment]
@situchan let us know how your testing went!

**Comment by muttmuure** [comment]
bump @situchan, just waiting on results from your testing: https://github.com/Expensify/App/issues/31276#issuecomment-1863202894

**Comment by situchan** [comment]
@dukenv0307 I am not sure how you will apply same logic to `UnreadIndicatorUpdater`

```
    const isHidden = optionItem.notificationPreference === CONST.REPORT.NOTIFICATION_PREFERENCE.HIDDEN;
    if (isHidden && !props.isFocused && !optionItem.isPinned) {
        return null;
    }
```

**Comment by situchan** [comment]
@dukenv0307's [proposal](https://github.com/Expensify/App/issues/31276#issuecomment-1809346822) looks good and tests well.
🎀 👀 🎀 C+ reviewed

**Comment by MonilBhavsar** [comment]
I see this is also being fixed in https://github.com/Expensify/App/pull/34025 as a regression from https://github.com/Expensify/App/pull/30133

**Comment by situchan** [comment]
ok, we can put this on hold

**Comment by muttmuure** [comment]
Held

**Comment by quinthar** [comment]
This feels like a wave5 issue as it deals with transaction threads; reassigning.

**Comment by muttmuure** [comment]
Still in review here: https://github.com/Expensify/App/pull/34025

**Comment by muttmuure** [comment]
Still in review

**Comment by muttmuure** [comment]
Still open

**Comment by muttmuure** [comment]
Not overdue!

**Comment by muttmuure** [comment]
Held

**Comment by MonilBhavsar** [comment]
PR merged and deployed to production. Taking this off hold.

**Comment by MonilBhavsar** [comment]
Have asked if bug is fixed or not in slack thread https://expensify.slack.com/archives/C049HHMV9SM/p1706532316571099?thread_ts=1699899029.212629&cid=C049HHMV9SM

**Comment by iwiznia** [comment]
This has been working for me for a while for me https://expensify.slack.com/archives/C03U7DCU4/p1706112016224459

**Comment by muttmuure** [comment]
It seems like there are a few PRs that will be eligible for payment here, I need to comb through and figure it out

**Comment by muttmuure** [comment]
OK, so the linked issue was a fix for a regression - https://github.com/Expensify/App/issues/33506 so no payment there. 

We did get a proposal for this issue above, but since there was an overlap with the regression fix, there was already something solid and we actually didn't need the second proposal. 

So I am going to close this, but if you think that your work was eligible for payment despite this issue not having a linked PR, please send me a DM!
